src/daft-parquet/src/python.rs

```diff
@@ -18,7 +18,7 @@ pub mod pylib {
     pub fn read_parquet(
         py: Python,
         uri: &str,
-        columns: Option<Vec<&str>>,
+        columns: Option<Vec<String>>,
```

Comment:
why does this have to be owned now?

Reply:
I'm not entirely sure why pyo3 doesn't allow `Vec<&str>` anymore but I suspect it's because of some sort of lifetime issue with the new Bound stuff